library/src/main/scala/sbt/SendEventService.scala

```diff
@@ -7,6 +7,20 @@ sealed trait SendEventService {
   def sendEvent[T: Pickler](event: T): Unit
 }
 
+object SendEventService {
+  private val detachedKey = AttributeKey[SendEventService]("SendEventService that sends DetachedEvent")
+
+  /**
+   * To send a detached plugin event, do getDetached(state).foreach(_.sendEvent(event)).
+   * This should never be done from inside a task or background job where a better
+   * SendEventService is available already. "Detached" means not associated with a task
+   * or job.
+   */
+  def getDetached(state: State): Option[SendEventService] = state get detachedKey
```

Comment:
We use "extract" in all other places in sbt (like Project.extract).  Not to nitpack, but we should be consistent, even if the original was odd.